### PR TITLE
Use EmptyResponse in swift code for no responses

### PR DESCRIPTION
**Pull request by roman-mazur**
@PaulTaykalo would you suggest me how to add a test? I do not even know if it compiles... :)

### CODE REVIEW

**Comment by roman-mazur** [comment]
@evgenbakumenko you might try this branch and see how it works for you...

**Comment by evgenbakumenko** [comment]
@roman-mazur works now

File: codegen/swift/swift-entities/src/main/resources/client/SwiftAPIClientCore.mustache [hunk 1]
```diff
@@ -117,6 +117,12 @@ public protocol Deserializable {
     static func deserialize(data: NSData?) -> Result<Self, NSError>
 }
 
+extension Void : Deserializable {
```
Comment (roman-mazur) [inline codegen/swift/swift-entities/src/main/resources/client/SwiftAPIClientCore.mustache]: @evgenbakumenko I'm a bit worried about this part: to my understanding such file can appear in multiple places in the codebase. Will it cause any issues or compilation errors - I do not know if it lives in its own namespace or scope?

Comment (evgenbakumenko) [inline codegen/swift/swift-entities/src/main/resources/client/SwiftAPIClientCore.mustache]: @roman-mazur you're right, got ```Type 'Void' (aka '()') has no member 'deserialize'``` compilation error now

**Comment by PaulTaykalo** [comment]
@roman-mazur I believe last time I used this one I was simply generating api and passed output to `swiftc` compiler. Unfortunately, this one was done locally. I believe at that moment `swiftc` wasn't working on the Travis CI correctly.
I believe You can add this one for tests - generate specific mappings with the script like here https://github.com/PaulTaykalo/helium-generation-example/blob/master/generate_api.sh
and then pass resulting .swift files to the `swiftc`

**Comment by PaulTaykalo** [comment]
@roman-mazur Also for what kind of the responses do you need to have `Void` as the result?

**Comment by roman-mazur** [comment]
@PaulTaykalo for those that have empty body, for instance DELETE request can respond with 204 and empty body
The spec, in such case, has no response definition

**Comment by PaulTaykalo** [comment]
@roman-mazur Can you give an example simple one of specs that contains these kind of responses?

**Comment by PaulTaykalo** [comment]
@roman-mazur found that swift-gen branch wasn't pushed 🤷‍♂️  take a look here https://github.com/stanfy/helium/tree/feature/swift-client-gen-test-example
at these lines especially
https://github.com/stanfy/helium/blob/feature/swift-client-gen-test-example/command-line/src/cmd-test/swift-gen-run.sh#L13

File: command-line/src/cmd-test/swift-gen-run.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+   -Fgenerate-equatables \
+   -Fgenerate-random-initializers \
+   -HentitiesAccessLevel=public \
+   -VprovidedVar=VariableValue \
+   src/cmd-test/test.api
+
+
+# this one should be ran (with 2.3 toolchain)
+# TODO: Add ReactiveCocoa module.
+# swiftc build/swift-gen-test-out/SwiftAPIClientCore.swift build/swift-gen-test-out/SwiftAPIRequestManager.swift build/swift-gen-test-out/SwiftAPIServiceExample.swift -o SwiftGenTestOut
```
Comment (roman-mazur) [inline command-line/src/cmd-test/swift-gen-run.sh]: I will work on travis toolchain in a follow-up, but we also need to resolve the TODO...

**Comment by roman-mazur** [comment]
I rebased all out changes into this PR, and it should be ready for merge.
@evgenbakumenko please confirm it will work for you

File: command-line/src/cmd-test/test.api [hunk 3]
```diff
@@ -11,4 +13,8 @@ service {
     response 'Entity'
   }
 
+  delete '/profile/{id}' spec {
+    name 'Delete profile'
```
Comment (roman-mazur) [inline command-line/src/cmd-test/test.api]: @PaulTaykalo it's a very simple example of supported method spec: response body is missing in this case.
You just get 204 with nothing.

**Comment by evgenbakumenko** [comment]
@roman-mazur yep, it's working

**Pull request merged**